### PR TITLE
Add product variant channel pricing gateway

### PR DESCRIPTION
**Pull request by rs2487**

### CODE REVIEW

File: Tests/Functional/FunctionalTestCate.php [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\SyliusConsumerBundle\Tests\Functional;
+
+use Sulu\Bundle\SyliusConsumerBundle\Tests\Service\GatewayClient;
+use Sulu\Bundle\TestBundle\Testing\SuluTestCase;
+use Symfony\Component\Messenger\MessageBusInterface;
+
+class FunctionalTestCate extends SuluTestCase
```
Comment (alexander-schranz) [inline Tests/Functional/FunctionalTestCate.php]: should be something like: `AbstractGatewayClientTestCase`

File: Resources/config/services/product-handler.xml [hunk 2]
```diff
@@ -124,5 +124,11 @@
 
             <tag name="messenger.message_handler"/>
         </service>
+
+        <service id="Sulu\Bundle\SyliusConsumerBundle\Model\Product\Handler\Query\LoadProductVariantChannelPricingQueryHandler">
```
Comment (alexander-schranz) [inline Resources/config/services/product-handler.xml]: @rs2487 you can let this as it is as its done before, but want to mention that this is bad practice in bundles using FCQN as ids. See https://symfony.com/doc/current/bundles/best_practices.html#services

> If the bundle defines services, they must be prefixed with the bundle alias. For example, AcmeBlogBundle services must be prefixed with acme_blog.

File: Resources/config/services/gateway.xml [hunk 3]
```diff
@@ -32,5 +32,10 @@
         <service id="Sulu\Bundle\SyliusConsumerBundle\Gateway\ProductAttributeGatewayInterface" class="Sulu\Bundle\SyliusConsumerBundle\Gateway\ProductAttributeGateway">
             <argument type="service" id="sulu_sylius_consumer.gateway_client" />
         </service>
+
+        <service id="Sulu\Bundle\SyliusConsumerBundle\Gateway\ProductVariantChannelPricingGatewayInterface"
```
Comment (alexander-schranz) [inline Resources/config/services/gateway.xml]: same here you can let it as it is but its [bad practice](https://symfony.com/doc/current/bundles/best_practices.html#services) to us fcqn as ids in bundles.

File: Tests/Functional/Model/Product/Handler/LoadProductVariantChannelPricingQueryTest.php [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+                     }'
+                );
+            }
+        );
+
+        $query = $query = new LoadProductVariantChannelPricingQuery('PRODUCT1', 'VARIANT1', 'channel-2');
+        $this->getMessageBus()->dispatch($query);
+
+        $channelPrice = $query->getPrice();
+        $this->assertEquals('1995', $channelPrice);
```
Comment (alexander-schranz) [inline Tests/Functional/Model/Product/Handler/LoadProductVariantChannelPricingQueryTest.php]: use `assertSame`

**Comment by chirimoya** [comment]
@rs2487 please fix comments than we can merge this PR

File: Tests/Functional/FunctionalTestCate.php (hunk 1 above)
Comment (rs2487) [reply Tests/Functional/FunctionalTestCate.php]: @alexander-schranz but there isnt any abstract method in it. So i think the `GatewayClientTestCase` would make more sense.

File: Resources/config/services/product-handler.xml (hunk 2 above)
Comment (rs2487) [reply Resources/config/services/product-handler.xml]: I agree with you by 100 % but i want be consistent with the existing IDs

File: Tests/Functional/Model/Product/Handler/LoadProductVariantChannelPricingQueryTest.php [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+                     }'
+                );
+            }
+        );
+
+        $query = $query = new LoadProductVariantChannelPricingQuery('PRODUCT1', 'VARIANT1', 'channel-2');
+        $this->getMessageBus()->dispatch($query);
+
+        $channelPrice = $query->getPrice();
+        $this->assertSame('1995', $channelPrice);
```
Comment (alexander-schranz) [inline Tests/Functional/Model/Product/Handler/LoadProductVariantChannelPricingQueryTest.php]: @rs2487 Test is failing returns your example json returns `1995` as int and not as string

**Comment by wachterjohannes** [comment]
ping @alexander-schranz @rs2487

**Comment by alexander-schranz** [comment]
@rs2487 merged

**Pull request merged**